main/src/ca/uwaterloo/flix/language/errors/StratificationError.scala

```diff
@@ -17,14 +17,33 @@
 package ca.uwaterloo.flix.language.errors
 
 import ca.uwaterloo.flix.language.CompilationError
+import ca.uwaterloo.flix.language.ast.TypedAst.Predicate.Head
+import ca.uwaterloo.flix.language.ast.TypedAst.Predicate.Head.{False, True}
 import ca.uwaterloo.flix.language.ast._
+import ca.uwaterloo.flix.util.vt.VirtualString._
 import ca.uwaterloo.flix.util.vt.VirtualTerminal
 
 /**
   * An error raised to indicate that a property is violated.
   */
-case class StratificationError(constraints: List[TypedAst.Constraint]) extends CompilationError {
+case class StratificationError(constrs: List[TypedAst.Constraint]) extends CompilationError {
   val kind: String = "Stratification Error"
-  val source: SourceInput = constraints.head.loc.source
-  val message: VirtualTerminal = ??? // TODO
+  val source: SourceInput = constrs.head.loc.source
+  val message: VirtualTerminal = {
+    val vt = new VirtualTerminal
+    vt << Line(kind, source.format) << NewLine
+    vt << ">> Stratification Error in Constraints" << NewLine
+
+    vt << (if (constrs.size > 1) {"Consider the rules:"} else {"Consider the rule:"}) << NewLine
+    constrs.foreach(rule => {
+      vt << (rule.head match {
+        case True(loc) => loc.lineAt(loc.beginLine)
```

Comment:
The problem with `loc.lineAt` is that a constraint may span more than one line. On the other hand, the `Code` formatter on `vt` requires some text message underneath each constraint. Perhaps the best solution is just to use `loc.format` which only prints the location (and not the actual source code).